packages/core/src/application.ts

```diff
@@ -161,6 +163,36 @@ export class Application extends Context {
     );
   }
 
+  // tslint:disable-next-line:no-any
+  private config(cfg: {[key: string]: any}) {
+    for (const key in cfg) {
+      const item = cfg[key];
+      // Iterate through the keys one level down (not recursively!)
+      if (item instanceof Object) {
+        for (const subkey in item) {
+          this.bindItemOrClass(`${key}.${subkey}`, item[subkey]);
+        }
+      }
+      this.bindItemOrClass(key, item);
+    }
+  }
+  /**
+   * Helper function for either binding an item or a class, depending.
+   * @private
+   * @param key The key of the configuration element.
+   * @param itemOrClass The item or class to bind.
+   */
+  // tslint:disable-next-line:no-any
+  private bindItemOrClass(key: string, itemOrClass: any) {
+    if (itemOrClass instanceof Function) {
+      this.bind(key)
+        .toClass(itemOrClass)
+        .inScope(BindingScope.SINGLETON);
```

Comment:
I am concerned this convention is very limiting. What if the configuration is expecting a regular function that should be bound via `this.bind.to(fn)`?

```ts
const config = {
  LoggerComponent: {
    writer: (msg: string) => saveLogToBackendViaHttpCall(msg)
  }
}
```

Reply:
What would be the best way to distinguish between a constructor and a regular function here? Additional checks for properties like `.constructor`?

Reply:
To my best knowledge, there is no way how to distinguish between a class constructor and a regular function, especially if the class constructor is using ES5 style.

Actually, if we are ok to enforce only ES6 `class` constructors, then it may be possible to detect classes by checking that `fn.toString()` starts with `class` keyword. Seee http://www.ecma-international.org/ecma-262/6.0/#sec-function.prototype.tostring.

```
$ node
> (class Foo {}).toString()
'class Foo {}'
```

However, are we sure that we are never going to want a configuration option that's expecting a class constructor instead of a class instance? I can imagine that for example an authentication component may want to receive custom User and AccessToken models (class constructors) in the configuration.

My point is that at framework level, it's not possible to reliable detect whether the user wants to bind function/class constructor via `.to()` or via `.toClass()` (or even `.toDynamicValue()`), we need the user to explicitly tell us their intent.